code/datums/records/record.dm

```diff
@@ -188,18 +206,29 @@
  * Returns an empty `/icon` if there was no `character_appearance` entry in the `fields` list,
  * returns the generated/cached photo otherwise.
  */
-/datum/record/crew/proc/get_photo(field_name, orientation)
+/datum/record/crew/proc/get_photo(field_name, orientation = SOUTH)
 	if(!field_name)
 		return
-
 	if(!character_appearance)
 		return new /icon()
+	var/obj/item/photo/existing_photo = LAZYACCESS(record_photos, field_name)
+	if(!existing_photo)
+		existing_photo = make_photo(field_name, orientation)
+	return existing_photo
 
+/**
+ * make_photo
+ *
+ * Called if the person doesn't already have a photo, this will make a photo of the person,
+ * then make a picture out of it, then finally create a new photo.
+ */
+/datum/record/crew/proc/make_photo(field_name, orientation, add_height_chart)
 	var/icon/picture_image
 	if(!isicon(character_appearance))
 		var/mutable_appearance/appearance = character_appearance
 		appearance.setDir(orientation)
-
+		if(add_height_chart)
+			appearance.underlays += mutable_appearance('icons/obj/machines/photobooth.dmi', "height_chart", alpha = 125)
```

Comment:
I think it'd be more efficient to add the height chart post `getFlatIcon`, so it doesn't get skewed by matrix transform or displacement map filters.

Reply:
Icons don't have underlays so I can't just mutable appearance it once it's been turned into one

Reply:
Very well, then at least do this:
```suggestion
		if(add_height_chart)
			appearance.underlays += mutable_appearance('icons/obj/machines/photobooth.dmi', "height_chart", alpha = 125, appearance_flags = RESET_TRANSFORM)
```

Reply:
Oh, you know, I should've also suggested the RESET_COLOR and RESET_ALPHA flags too. My bad for bothering you twice.